gammapy/scripts/make_bg_cube_models.py

```diff
@@ -0,0 +1,321 @@
+
+    1. make a global event list from a datastore
+    2. filter the runs keeping only the ones far from known sources
+    3. group the runs according to similar observation conditions (i.e. alt, az)
+        * using `~gammapy.obs.ObservationGroups`
+    4. create a bg cube model for each group using:
+        * the `~gammapy.background.make_bg_cube_model` method
+        * and `~gammapy.background.CubeBackgroundModel` objects as containers
+
+    The models are stored into FITS files.
+
+    It can take a few minutes to run. For a quicker test, please activate the
+    **test** flag.
+
+    TODO: revise doc!!!
+
+    Parameters
+    ----------
+    fitspath : str
+        Path to dir containing list of input fits event files.
+    scheme : str
+        Scheme of file naming.
+    outdir : str
+        Dir path to store the results.
+    overwrite : bool
+        If true, run fast (not recomended for analysis).
+    test : bool
+        If true, run fast (not recomended for analysis).
+    a_la_michi : bool, optional
+        Flag to activate Michael Mayer's bg cube production method.
+
+    Examples
+    --------
+    >>> gammapy-make-bg-cube-models -h
+    >>> gammapy-make-bg-cube-models /path/to/fits/event_lists/base/dir HESS bg_cube_models
+    >>> gammapy-make-bg-cube-models /path/to/fits/event_lists/base/dir HESS bg_cube_models --test
+    >>> gammapy-make-bg-cube-models /path/to/fits/event_lists/base/dir HESS bg_cube_models --test --overwrite
+    >>> gammapy-make-bg-cube-models /path/to/fits/event_lists/base/dir HESS bg_cube_models --a-la-michi
+
+    """
+    # create output folder
+    if not os.path.isdir(outdir):
+        os.mkdir(outdir)
+    else:
+        if overwrite:
+            # delete and create again
+            shutil.rmtree(outdir) # recursively
+            os.mkdir(outdir)
+        else:
+            # do not overwrite, hence exit
+            s_error = "Cannot continue: directory \'{}\' exists.".format(outdir)
+            raise RuntimeError(s_error)
+
+    create_bg_observation_list(fitspath, scheme, outdir, overwrite, test)
+    group_observations(outdir, overwrite, test)
+    stack_observations(fitspath, outdir, overwrite, a_la_michi)
+
+
+def create_bg_observation_list(fits_path, scheme, outdir, overwrite, test):
+    """Make total observation list and filter the observations.
+
+    In a first version, all obs taken within 3 deg of a known source
+    will be rejected. If a source is extended, twice the extension is
+    added to the corresponding exclusion region radius of 3 deg.
+
+    Parameters
+    ----------
+    fits_path : str
+        Path to dir containing list of input fits event files.
+    scheme : str
+        Scheme of file naming.
+    outdir : str
+        Dir path to store the results.
+    overwrite : bool
+        If true, run fast (not recomended for analysis).
+    test : bool
+        If true, run fast: skip many runs and catalog sources.
+    """
+    print()
```

Comment:
Maybe log like this to avoid having the first line indented and add newlines to make it more readable if you want?
```python
# Just an example, not really meant to insert this here
log.info('\nObservation table:\n{}\n'.format(table))
```
This way we get nicer output, but the user still has control over where the log output goes.

If this is still not good for some reason, feel free to make a new issue so that we can try to find a better solution.